From 2a6ce26e91f1ab42d314d41b8c28bef64df50a2e Mon Sep 17 00:00:00 2001
From: Alexander Stefanov-Khryukin <alexander@mezon.ru>
Date: Tue, 29 Nov 2016 14:09:11 +0000
Subject: [PATCH] Too many things don't get along with the unified hierarchy
 yet:

 * opencontainers/runc#1175
 * docker/docker#28109
 * lxc/lxc#1280

So revert the default to the legacy hierarchy for now. Developers of the
above
software can opt into the unified hierarchy with
"systemd.legacy_systemd_cgroup_controller=0".
(cherry picked from commit 843d5baf6aad6c53fc00ea8d95d83209a4f92de1)
---
 src/basic/cgroup-util.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/basic/cgroup-util.c b/src/basic/cgroup-util.c
index cede835..134e6e3 100644
--- a/src/basic/cgroup-util.c
+++ b/src/basic/cgroup-util.c
@@ -2423,10 +2423,10 @@ bool cg_is_unified_systemd_controller_wanted(void) {
 
                 r = get_proc_cmdline_key("systemd.legacy_systemd_cgroup_controller=", &value);
                 if (r < 0)
-                        return true;
+                        return false;
 
                 if (r == 0)
-                        wanted = true;
+                        wanted = false;
                 else
                         wanted = parse_boolean(value) <= 0;
         }
-- 
2.8.3

